test/new-e2e/tests/process/fargate_test.go

```diff
@@ -0,0 +1,121 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024-present Datadog, Inc.
+
+package process
+
+import (
+	"testing"
+	"time"
+
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/runner"
+	"github.com/DataDog/test-infra-definitions/components/datadog/apps/cpustress"
+	"github.com/DataDog/test-infra-definitions/resources/aws"
+	"github.com/pulumi/pulumi/sdk/v3/go/auto"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+	"github.com/stretchr/testify/assert"
+
+	fakeintakeComp "github.com/DataDog/test-infra-definitions/components/datadog/fakeintake"
+	ecsComp "github.com/DataDog/test-infra-definitions/components/ecs"
+
+	"github.com/DataDog/datadog-agent/pkg/util/testutil/flake"
+	"github.com/DataDog/datadog-agent/test/fakeintake/aggregator"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/e2e"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/environments"
+
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/environments/aws/ecs"
+)
+
+type ECSFargateSuite struct {
+	e2e.BaseSuite[fargateCPUStressEnv]
+}
+
+type fargateCPUStressEnv struct {
+	environments.ECS
+}
+
+func ecsFargateCPUStressProvisioner() e2e.PulumiEnvRunFunc[fargateCPUStressEnv] {
+	return func(ctx *pulumi.Context, env *fargateCPUStressEnv) error {
+		awsEnv, err := aws.NewEnvironment(ctx)
+		if err != nil {
+			return err
+		}
+
+		params := ecs.GetProvisionerParams(
+			ecs.WithAwsEnv(&awsEnv),
+			ecs.WithECSFargateCapacityProvider(),
+			ecs.WithFargateWorkloadApp(func(e aws.Environment, clusterArn pulumi.StringInput, apiKeySSMParamName pulumi.StringInput, fakeIntake *fakeintakeComp.Fakeintake) (*ecsComp.Workload, error) {
+				return cpustress.FargateAppDefinition(e, clusterArn, apiKeySSMParamName, fakeIntake)
+			}),
+		)
+
+		if err := ecs.Run(ctx, &env.ECS, params); err != nil {
+			return err
+		}
+
+		return nil
+	}
+}
+
+func TestECSFargateTestSuite(t *testing.T) {
+	t.Parallel()
+	s := ECSFargateSuite{}
+	e2eParams := []e2e.SuiteOption{e2e.WithProvisioner(
+		e2e.NewTypedPulumiProvisioner("ecsFargateCPUStress", ecsFargateCPUStressProvisioner(), nil),
+	),
+	}
+
+	e2e.Run(t, &s, e2eParams...)
+}
+
+func (s *ECSFargateSuite) TestProcessCheck() {
+	t := s.T()
+	// PROCS-4219
+	flake.Mark(t)
```

Comment:
is this supposed to be marked flaky right off the bat? 🤔

Reply:
This is on purpose as the original ECS EC2 tests I based these off are flaky. I did increase the timeout for fake intake to see if it would help. I did this to avoid being pinged in case it isnt fixed 😓